### PR TITLE
Fixing node-agent socket problem after temporary network issue

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
tendrl-bug-id: Tendrl/node-agent#850
bugzilla: 1630344

Signed-off-by: GowthamShanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

**Comment by GowthamShanmugam** [comment]
@shtripat @nthomas-redhat @anmolsachan please review

File: tendrl/node_agent/message/handler.py [hunk 1]
```diff
@@ -90,15 +90,19 @@ def bind_unix_listener(self):
         # http://0pointer.de/blog/projects/systemd.html (search "file
         # descriptor 3")
         try:
-            socket_fd = 3
-            self.sock = socket.fromfd(socket_fd, socket.AF_UNIX,
-                                      socket.SOCK_STREAM)
-            self.sock.listen(50)
-            return self.sock
+            # systemd socket should exist before listen
+            if os.path.exists(MESSAGE_SOCK_PATH):
```
Comment (shtripat) [inline tendrl/node_agent/message/handler.py]: Who creates this file? Sometimes we have seen the socket file doesnt exist, even if tendrl-node-agent is running. How we tackle that scenario?

Comment (GowthamShanmugam) [reply tendrl/node_agent/message/handler.py]: message.socket file is created by node-agent.socket systemd file, node-agent systemd file is invoking that. The reason which we are not seen message socket is sometime node-agent start won't invoking node-agent.socket systemd file but node-agent restart will invoke this always.

When connection issue happening with etcd then node-agent will keep trying to connect after few minutes node-agent will start again(this scenario I saw in a customer machine) so node-agent start is not invoking node-agent.socket systemd so message.sock file is not created. 

As per this patch if message.sock file does not present then below code https://github.com/Tendrl/node-agent/pull/851/files#diff-3eae70b9dbe53a810f8c236acc9baa6dR103 will tack care of creating. So message.sock file will always present.

Comment (shtripat) [reply tendrl/node_agent/message/handler.py]: Ok. Looks fine.

**Comment by GowthamShanmugam** [comment]
@shtripat i agree

**Comment by GowthamShanmugam** [comment]
@nthomas-redhat please review and merge this PR

**Pull request merged**